drivers/sms_ser.c

```diff
@@ -0,0 +1,698 @@
+        } else {
+            snprintfcat(results->version, 6, "%c", rawvalues[i]);
+        }
+    }
+}
+
+void sms_parse_results(uint8_t *rawvalues, SmsData *results) {
+    char buf[BUFFER_SIZE];
+    unsigned char byte, mask, p;
+    long v;
+    double h;
+
+    memset(buf, 0, BUFFER_SIZE);
+    sprintf(buf, "0x%02x%02x", (unsigned char)rawvalues[1], (unsigned char)rawvalues[2]);
+    v = strtol(buf, NULL, 16); /* 16 == hex */
+    h = v / 10;
+    results->lastinputVac = h;
+
+    memset(buf, 0, BUFFER_SIZE);
+    sprintf(buf, "0x%02x%02x", (unsigned char)rawvalues[3], (unsigned char)rawvalues[4]);
+    v = strtol(buf, NULL, 16); /* 16 == hex */
+    h = v / 10;
+    results->inputVac = h;
+
+    memset(buf, 0, BUFFER_SIZE);
+    sprintf(buf, "0x%02x%02x", (unsigned char)rawvalues[5], (unsigned char)rawvalues[6]);
+    v = strtol(buf, NULL, 16); /* 16 == hex */
+    h = v / 10;
+    results->outputVac = h;
+
+    memset(buf, 0, BUFFER_SIZE);
+    sprintf(buf, "0x%02x%02x", (unsigned char)rawvalues[7], (unsigned char)rawvalues[8]);
+    v = strtol(buf, NULL, 16); /* 16 == hex */
+    h = v / 10;
+    results->outputpower = h;
+
+    memset(buf, 0, BUFFER_SIZE);
+    sprintf(buf, "0x%02x%02x", (unsigned char)rawvalues[9], (unsigned char)rawvalues[10]);
+    v = strtol(buf, NULL, 16); /* 16 == hex */
+    h = v / 10;
+    results->outputHz = h;
+
+    memset(buf, 0, BUFFER_SIZE);
+    sprintf(buf, "0x%02x%02x", (unsigned char)rawvalues[11], (unsigned char)rawvalues[12]);
+    v = strtol(buf, NULL, 16); /* 16 == hex */
+    h = v / 10;
+    results->batterylevel = h;
+
+    memset(buf, 0, BUFFER_SIZE);
+    sprintf(buf, "0x%02x%02x", (unsigned char)rawvalues[13], (unsigned char)rawvalues[14]);
+    v = strtol(buf, NULL, 16); /* 16 == hex */
+    h = v / 10;
+    results->temperatureC = h;
+
+    byte = rawvalues[15];
+    p = rawvalues[15];
+    mask = 1;
+
+    results->beepon = ((byte & (mask << 0)) != 0) ? true : false;
+    results->shutdown = ((byte & (mask << 1)) != 0) ? true : false;
+    results->test = ((byte & (mask << 2)) != 0) ? true : false;
+    results->upsok = ((byte & (mask << 3)) != 0) ? true : false;
+    results->boost = ((byte & (mask << 4)) != 0) ? true : false;
+    results->bypass = ((byte & (mask << 5)) != 0) ? true : false;
+    results->lowbattery = ((byte & (mask << 6)) != 0) ? true : false;
+    results->onbattery = ((byte & (mask << 7)) != 0) ? true : false;
+
+    upslogx(LOG_INFO, "beepon: %d", results->beepon);
+    upslogx(LOG_INFO, "shutdown: %d", results->shutdown);
+    upslogx(LOG_INFO, "test: %d", results->test);
+    upslogx(LOG_INFO, "upsok: %d", results->upsok);
+    upslogx(LOG_INFO, "bypass: %d", results->bypass);
+    upslogx(LOG_INFO, "lowbattery: %d", results->lowbattery);
+    upslogx(LOG_INFO, "onbattery: %d", results->onbattery);
+}
+
+static int get_ups_nominal() {
+    uint8_t length;
+    ssize_t ret;
+
+    upslogx(LOG_INFO, "get_ups_nominal");
```

Comment:
Here and in other places, `upslogx` may be too loud (and going to syslog etc.) Consider changing to `upsdebugx` with a non-zero verbosity level, at least where it is "just" debug tracing.

Reply:
Ok ! A lot of things was to "ok, i get to this part", i will review all logs.